esmvalcore/_recipe.py

```diff
@@ -903,6 +903,8 @@ def _initialize_variables(self, raw_variable, raw_datasets):
             institute = get_institutes(variable)
             if institute:
                 variable['institute'] = institute
+            if 'CMIP6' in variable['project']:
```

Comment:
I would make this a bit more general, e.g. similar to how institute is retrieved above: if you can find an activity in the projects cmor table, add it, if not leave it out.